Modules/socketmodule.c

```diff
@@ -7922,6 +7922,9 @@ socket_exec(PyObject *m)
     ADD_INT_MACRO(m, SO_OOBINLINE);
 #endif
 #ifndef __GNU__
+#ifdef  SO_ORIGINAL_DST
```

Comment:
Is it only exposed if `__GNU__` is not defined?

Reply:
Under non-standard `GNU/linux`, this macro may not exist in linux. It's right.

Reply:
But it's not defined like this in linux/netfilter_ipv4.h: https://github.com/torvalds/linux/blob/27cc6fdf720183dce1dbd293483ec5a9cb6b595e/include/uapi/linux/netfilter_ipv4.h#L52.

And it shouldn't be an issue since we already do `#ifdef  SO_ORIGINAL_DST`. So we don't really care. Note that this macro is only used twice in the entire code base, and only for SO_REUSEPORT in sockets, so I'm really wondering whether this should be under this if-guard.

Reply:
> I'm sorry, I didn't see it. In non-standard GNU/linux, this macro may have been defined by other libraries (because this's in netfilter, not defined by glibc).

But we don't care about the other libraries since we include the "linux/netfilter_ipv4.h" header. So it will be defined by the latter. None of the other constants have this specific treatment as well.

Reply:
> But it's not defined like this in linux/netfilter_ipv4.h: https://github.com/torvalds/linux/blob/27cc6fdf720183dce1dbd293483ec5a9cb6b595e/include/uapi/linux/netfilter_ipv4.h#L52.

Hmmm, But, Torvalds's tree is **GNU/Linux**. This may refer to the user's secondary distribution of Linux.

Reply:
Other constants don't care about this. We usually support platforms as defined by PEP-11 and we are not fool-proof against custom distributions. Honestly unless there is a real reason **why** we are hiding it under the `#ifndef __GNU__` guard (which, by the way, would actually indicate that the platform is GNU-friendly if it were present). 

Generally `__GNU__` and co (such as `__APPLE__`, `__FreeBSD__`, etc) are mainly used to detect how to include header files, e.g.:

```C
#if (defined(__APPLE__) && defined(__MACH__)) || defined(__linux__)
#include <string.h>
#endif

#if defined(__FreeBSD__) || defined(__NetBSD__)
#include <strings.h>
#endif
```

So it's more for a platform-specific implementation. However, we already know that we are including `linux/netfilter_ipv4.h`.

Reply:
Ill take it out of the ifndef

Reply:
Could you explain your initial rationale on why you put the macro under this specific guard? is there a possibility where this macro is defined differently or means something else, e.g., on *BSD systems?

Reply:
Ok, that's enough for me. Thanks for the reply!